node/src/accountant/scanners.rs

```diff
@@ -331,13 +373,17 @@ impl Scanner<RequestTransactionReceipts, ReportTransactionReceipts> for PendingP
         message: ReportTransactionReceipts,
         logger: &Logger,
     ) -> Option<NodeToUiMessage> {
-        debug!(
-            logger,
-            "Processing receipts for {} transactions",
-            message.fingerprints_with_receipts.len()
-        );
-        let statuses = self.handle_pending_transaction_with_its_receipt(&message, logger);
-        self.process_transaction_by_status(statuses, logger);
+        if message.fingerprints_with_receipts.len() != 0 {
```

Comment:
I find `!message.fingerprints_with_receipts.is_empty()` nicer and clearer but that's your decision...